File: corvus-app/src/main/java/by/varyvoda/corvus/app/model/source/FileSource.java
package by.varyvoda.corvus.app.model.source;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
@DiscriminatorValue("FILE")
public class FileSource extends Source {

    {
        setType(Type.FILE);
    }

    @Column(name = "content")
    private byte[] content;

}

File: client/src/app/base/models/source/file-source.ts
import {Source} from "./source";

export interface FileSource extends Source<"FILE"> {
    name: string;
    loadingProgress: number;
}

export const createFileSource = (file: File, progress: number): FileSource => ({
    id: -1,
    type: "FILE",
    name: file.name,
    loadingProgress: progress,
});
